odk1-src/form-logic.rst

```diff
@@ -0,0 +1,456 @@
+:tc:`contains(${haystack}, 'needle')`
+  True if the exact string ``needle`` 
+  is contained anywhere inside the response to :tc:`haystack`.
+  
+:tc:`count-selected(${toppings}) > 5`
+  True if more than five options were selected
+  in the :ref:`multi-select-widget` named :tc:`toppings`.
+
+.. _XPath functions: https://opendatakit.github.io/xforms-spec/#xpath-functions
+
+
+.. _repeats:
+
+Repeating questions and groups of questions
+==============================================
+
+To repeat questions or groups of questions
+use the :tc:`begin_repeat...end_repeat` syntax.
+This can be used with one question or a group of questions.
+
+.. rubric:: XLSForm
+
+.. csv-table:: survey
+  :header: type, name, label 
+
+  begin_repeat, my_repeat_group, Repeat group label
+  text, repeated_question, This question will be repeated.
+  end_repeat, , 
+
+.. rubric:: XLSForm
+
+.. csv-table:: survey
+  :header: type, name, label 
+
+  begin_repeat, my_repeat, Repeat group label
+  note, repeated_note, These questions will be repeated as an entire group.
+  text, name, What is your name?
+  text, quest, What is your quest?
+  text, fave_color, What is your favorite color?
+  end_repeat, , 
+ 
+  
+.. _controlling-number-of-repeats:
+
+Controlling the number of repeats
+------------------------------------
+    
+.. _user-controlled-repeats:
+    
+User controlled repeats
+""""""""""""""""""""""""""
+
+By default,
+the user controls how many times 
+the questions are repeated.
+
+At the end of each repetition,
+the user is asked if they want to add another repeat group.
+
+.. note::
+
+  The :th:`label` in the :tc:`begin_repeat` row
+  is shown in the **Add New Group?** message.
+  
+  A meaningful label will help enumerators and participants 
+  navigate the form as intended.
+
+.. image of this
+
+.. _statically-defined-repeats:
+
+Statically defined repeats
+----------------------------
+
+Use the :th:`repeat_count` columns
+to define the number of times a group will repeat.
+
+
+.. rubric:: XLSForm
+
+.. csv-table:: survey
+  :header: type, name, label, repeat_count
+
+  begin_repeat, my_repeat, Repeat group label, 3
+  note, repeated_note, These questions will be repeated as an entire group.
+  text, name, What is your name?
+  text, quest, What is your quest?
+  text, fave_color, What is your favorite color?
+  end_repeat, , 
+ 
+.. _dynamically-defined-repeats:
+ 
+Dynamically defined repeats
+""""""""""""""""""""""""""""
+
+The :th:`repeat_count` column can reference
+:ref:`previous responses <variables>` and :ref:`calculations <calculations>`.
+
+.. rubric:: XLSForm
+
+.. csv-table:: survey
+  :header: type, name, label, repeat_count
+  
+  integer, number_of_children, How many children do you have?
+  begin_repeat, child_questions, Questions about child, ${number_of_children}
+  text, child_name, Child's name,
+  integer, child_age, Child's age,
+  end_repeat, , , 
+
+  
+.. _cascading-selects:
+  
+Filtering options based on previous responses
```

Comment:
I don't think the `based on previous responses` is needed here because validating and restricting options (one of the previous sections) also has the property.